deeppavlov/skills/dsl_skill/utils.py

```diff
@@ -11,13 +11,51 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-
-from typing import Union, NamedTuple
+import logging
+import sqlite3
+from typing import Union, NamedTuple, Optional, Iterable
 
 UserId = Union[str, int]
 
 
 class SkillResponse(NamedTuple):
     response: str
     confidence: float
+
+
+class DummyLogger(logging.Logger):
+    """
+    Logger that does nothing.
+    """
+
+    def _log(self, level, msg, args, exc_info=None, extra=None, stack_info=False):
+        return None
+
+
+def execute_query(connection_string: str,
+                  sql_query: str,
+                  params: Optional[Iterable] = None,
+                  logger: Optional[logging.Logger] = None):
+    logger = logger or DummyLogger('dummy')
+    if connection_string:
+        connection = sqlite3.connect(connection_string)
+        try:
+            cursor = connection.cursor()
+            if params:
```

Comment:
maybe:
```
cursor.execute(sql_query, params or ())
```

Reply:
We get a strange construction when we try to reconnect: if we successfully reconnect, we call execute_query() one more time.

Reply:
connect first if successful for a limited number of attempts, run execute()